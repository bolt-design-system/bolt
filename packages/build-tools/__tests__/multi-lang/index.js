const path = require('path');
const fs = require('fs-extra');
const prettier = require('prettier');
const prettierConfig = { semi: false, parser: 'css' };

const buildDir = require('./.boltrc.multi-lang.js').buildDir;

describe('Tests to confirm that the latest upcoming release of the @bolt/build-tools is compiling as expected', () => {
  test('Check that Bolt Build Tools is generating JSON data from Sass files', async () => {
    const dataFileExists = async function() {
      try {
        await fs.promises.access(
          path.resolve(
            process.cwd(),
            `${buildDir}/data/colors/brand.bolt.json`,
          ),
        );
        return true;
      } catch (error) {
        return false;
      }
    };

    expect(await dataFileExists()).toBe(true);
  });

  test('Check JSON data for colors was generated and outputted by the @bolt/build-tools', async () => {
    const getColorData = async function() {
      try {
        await fs.promises.access(
          path.resolve(
            process.cwd(),
            `${buildDir}/data/colors/brand.bolt.json`,
          ),
        );
        const colorData = JSON.parse(
          fs.readFileSync(
            path.resolve(
              process.cwd(),
              `${buildDir}/data/colors/brand.bolt.json`,
            ),
          ),
        );
        return colorData;
      } catch (error) {
        return false;
      }
    };

    const colorData = await getColorData();

    expect(colorData).toMatchSnapshot();

    expect(colorData.navy.xdark).toBe('rgb(9, 23, 52)');
    expect(colorData.yellow.base).toBe('rgb(255, 200, 54)');
  });

  test('Check that the Japanese-specific CSS bundle was generated and contains the correct font', async () => {
    const getCSSData = async function() {
      try {
        const cssData = fs
          .readFileSync(
            path.resolve(process.cwd(), `${buildDir}/bolt-global-ja.css`),
          )
          .toString();
        return cssData;
      } catch (error) {
        return false;
      }
    };

    const cssFile = await getCSSData();

    const japaneseFontRule =
      'font-family:-apple-system,BlinkMacSystemFont,ヒラギノ角ゴ ProN,Hiragino Kaku Gothic ProN,游ゴシック,游ゴシック体,YuGothic,Yu Gothic,メイリオ,Meiryo,ＭＳ ゴシック,MS Gothic,HiraKakuProN-W3,TakaoExゴシック,TakaoExGothic,MotoyaLCedar,Droid Sans Japanese,sans-serif';

    const englishFontFamilyRule =
      'font-family:Open Sans,Helvetica Neue,sans-serif;font-family:var(--bolt-font-family-heading)';

    expect(cssFile).toEqual(expect.stringContaining(japaneseFontRule));
    expect(cssFile).toEqual(expect.not.stringContaining(englishFontFamilyRule));

    const prettyCss = await prettier.format(cssFile, prettierConfig);
    expect(prettyCss).toMatchSnapshot();
  });

  test('Check that the Japanese-specific bolt-global-ja.css file built contains the correct utility class styles for the older u-bolt-margin-bottom syntax.', async () => {
    const getCSSData = async function() {
      try {
        const cssData = fs
          .readFileSync(
            path.resolve(process.cwd(), `${buildDir}/bolt-global-ja.css`),
          )
          .toString();
        return cssData;
      } catch (error) {
        return false;
      }
    };

    const cssFile = await getCSSData();
    const prettyCss = await prettier.format(cssFile, prettierConfig);

    expect(prettyCss).toEqual(
      expect.stringContaining(
        await prettier.format(
          '.u-bolt-margin-bottom{margin-bottom:1.155rem!important}',
          prettierConfig,
        ),
      ),
    );
  });

  test('Check that the default English-specific bolt-global.css file built contains the correct utility class styles for the older u-bolt-margin-bottom syntax.', async () => {
    const getCSSData = async function() {
      try {
        const cssData = fs
          .readFileSync(
            path.resolve(process.cwd(), `${buildDir}/bolt-global-en.css`),
          )
          .toString();
        return cssData;
      } catch (error) {
        return false;
      }
    };

    const cssFile = await getCSSData();
    const prettyCss = await prettier.format(cssFile, prettierConfig);

    expect(prettyCss).toEqual(
      expect.stringContaining(
        await prettier.format(
          `.u-bolt-margin-bottom {margin-bottom: 1.35rem !important;}`,
          prettierConfig,
        ),
      ),
    );

    expect(prettyCss).toMatchSnapshot();
  });
});
