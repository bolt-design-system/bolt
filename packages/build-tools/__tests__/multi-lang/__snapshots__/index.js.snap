// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to confirm that the latest upcoming release of the @bolt/build-tools is compiling as expected Check JSON data for colors was generated and outputted by the @bolt/build-tools 1`] = `
Object {
  "black": Object {
    "base": "rgb(21, 22, 25)",
  },
  "gray": Object {
    "base": "rgb(141, 142, 153)",
    "dark": "rgb(92, 93, 102)",
    "light": "rgb(224, 226, 235)",
    "xdark": "rgb(46, 46, 51)",
    "xlight": "rgb(246, 246, 249)",
  },
  "indigo": Object {
    "base": "rgb(0, 31, 95)",
    "dark": "rgb(11, 29, 65)",
    "light": "rgb(62, 103, 187)",
    "xdark": "rgb(9, 23, 52)",
    "xlight": "rgb(171, 194, 242)",
  },
  "navy": Object {
    "base": "rgb(0, 31, 95)",
    "dark": "rgb(11, 29, 65)",
    "light": "rgb(62, 103, 187)",
    "xdark": "rgb(9, 23, 52)",
    "xlight": "rgb(171, 194, 242)",
  },
  "orange": Object {
    "base": "rgb(247, 105, 35)",
    "dark": "rgb(181, 72, 18)",
    "light": "rgb(255, 152, 102)",
    "xdark": "rgb(102, 34, 0)",
    "xlight": "rgb(255, 204, 179)",
  },
  "teal": Object {
    "base": "rgb(16, 165, 172)",
    "dark": "rgb(10, 98, 102)",
    "light": "rgb(134, 202, 198)",
    "xdark": "rgb(0, 81, 84)",
    "xlight": "rgb(183, 228, 230)",
  },
  "white": Object {
    "base": "rgb(255, 255, 255)",
  },
  "yellow": Object {
    "base": "rgb(255, 200, 54)",
    "dark": "rgb(232, 175, 23)",
    "light": "rgb(255, 227, 153)",
    "xdark": "rgb(102, 74, 0)",
    "xlight": "rgb(255, 241, 204)",
  },
}
`;
